src/include/cell_inline.h

```diff
@@ -96,11 +84,20 @@ __cell_pack_value_validity(WT_SESSION_IMPL *session, uint8_t **pp, WT_TIME_WINDO
         LF_SET(WT_CELL_TXN_START);
     }
 
-    /* If we write prepare_ts to start_ts, we write prepared_id to durable_start_ts as well */
     if (pack_prepare_info_to_start) {
-        WT_ASSERT(session, tw->start_prepared_id != WT_PREPARED_ID_NONE);
-        WT_RET(__wt_vpack_uint(pp, 0, tw->start_prepared_id));
-        LF_SET(WT_CELL_TS_DURABLE_START);
+        /*
+         * If the preserve prepared config is enabled, we write prepared_id to durable_start_ts as
+         * well
+         */
+        if (F_ISSET(S2C(session), WT_CONN_PRESERVE_PREPARED)) {
+            WT_ASSERT(session, tw->start_prepared_id != WT_PREPARED_ID_NONE);
+            WT_RET(__wt_vpack_uint(pp, 0, tw->start_prepared_id));
+            LF_SET(WT_CELL_TS_DURABLE_START);
+        } else if (tw->start_prepare_ts - reference_ts > 0) {
```

Comment:
I didn't understand clearly in what scenario we come here.
1. Before preserve config, we write the prepare already at the `WT_CELL_TS_START`. There is no durable ts or anything to write here.
2. With this preserve config, we wrote the prepare above, the prepare_id with `WT_CELL_TS_DURABLE_START`.

Can you explain this case?

Reply:
done